src/common/snippets/src/op/subgraph.cpp

```diff
@@ -71,6 +75,11 @@ void snippets::op::Subgraph::validate_and_infer_types() {
     for (size_t i = 0; i < get_output_size(); ++i) {
         set_output_type(i, m_body->get_output_element_type(i), m_body->get_output_partial_shape(i));
     }
+
+    const auto ops = m_body->get_ops();
+    m_is_quantized = std::any_of(ops.cbegin(), ops.cend(), [](const std::shared_ptr<Node>& op) {
+                                                                return ov::is_type<ov::op::v0::FakeQuantize>(op);
+                                                            });
```

Comment:
It's not clear why do we need to check for FQ every time we call `validate_and_infer_types()`, since only a couple of transformation can change the number of FQs. Maybe it's better to update `m_is_quantized` in the corresponding passes.

Reply:
Totally agree. Good catch! Thank you very much. I moved it into ctor.